### ISSUE TITLE
too-many-function-args not raised with list.extend()

### ISSUE THREAD

**Issue by pyoor** [PyCQA/pylint#3926]
It appears that pylint fails to detect invalid number of arguments passed to the list `extend` method.

```
>>> x = []
>>> x.extend(1,2,3)
```

### Current behavior
No error is raised.

### Expected behavior
This should raise a `too-many-function-args` error.

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.6 (default, Oct  6 2020, 03:22:36) 
[GCC 7.5.0]
```

**Comment by hippo91** [comment]
@pyoor thanks for your report. I can reproduce it.

**Comment by ethan-leba** [comment]
It looks like this is a symptom of a larger problem, which is that pylint is [incapable of type checking any built-in functions](https://github.com/PyCQA/pylint/blob/master/pylint/checkers/typecheck.py#L1248).

**Comment by hippo91** [comment]
@ethan-leba the problem here is not specific to `python`'s built in functions.
In fact `pylint`, which is a **static** linter, analyzes the content of `python` source files without executing them. Thus, if some functions/methods are dynamically constructed and especially those which are `c` object, then `pylint` can not deduce anything from them.
It is the case here as `extend` method is in fact coded inside `cpython`'s `./Objects/listobject.c` file.
